docs/src/main/asciidoc/funqy.adoc

```diff
@@ -0,0 +1,140 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - Funqy
+
+include::./attributes.adoc[]
+:extension-status: preview
+
+Quarkus Funqy is part of Quarkus's serverless strategy and aims to provide a portable Java API to write functions
+deployable to various FaaS environments like AWS Lambda, Azure Functions, Knative, and Knative events.  It is also
+usable outside of FaaS in more traditional environments.
+
+Another goal of Funqy is to create a remoting, RPC framework that is as small and optimized as possible for the
+Quarkus runtime.  This means sacrificing on flexibility to provide a runtime that has little to no overhead.  Funqy
+should never become more complicated than you see in this initial doc.
+
+== Funqy Basics
+
+The Funqy API is simple.  Annotate a method with `@Funq`.  This method may only have one optional input parameter
+and may or may not return a response.
+
+[source, java]
+----
+import io.quarkus.funqy.Funq;
+
+public class GreetingFunction {
+    @Funq
+    public String greet(String name) {
+       return "Hello " + name;
+    }
+}
+----
+
+Java classes can also be used as input and output and must follow the Java bean convention and have
```

Comment:
With Java records around the corner (preview in 14), there might be some push-back on requiring JavaBeans convention. Would be nice to see supports for record types here eventually (immutable, foo() instead of getFoo(), constructor expecting all components).

Reply:
Many companies are still on Java 8, so, I pushback your pushback!

Reply:
Hehe; well, even Quarkus is considering to move to 11 as a baseline, so it can't be that bad. Anyways, just meant it as nice future improvement; Jackson is already working on this, so you might get it for free even and only docs would need relaxing eventually.